tests/unit/AgentBuilderTest.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+use AG\ElasticApmLaravel\Agent;
+use AG\ElasticApmLaravel\AgentBuilder;
+use AG\ElasticApmLaravel\Collectors\SpanCollector;
+use AG\ElasticApmLaravel\Exception\MissingAppConfigurationException;
+use Codeception\Test\Unit;
+use Illuminate\Config\Repository as Config;
+
+class AgentBuilderTest extends Unit
+{
+    use Codeception\AssertThrows;
+
+    /** @var AgentBuilder */
+    private $builder;
+
+    /** @var SpanCollector */
+    private $collector;
+
+    protected function _before(): void
+    {
+        $this->builder = new AgentBuilder();
+        $this->collector = Mockery::mock(SpanCollector::class);
+    }
+
+    public function testMissingConfigurationException(): void
+    {
+        self::assertThrows(MissingAppConfigurationException::class, function () {
+            $this->builder->build();
+        });
+    }
+
+    public function testBuildAgent()
+    {
+        /** @var Agent */
+        $agent = $this->builder
+            ->withAppConfig(new Config(['serviceName' => 'Test']))
+            ->build();
+
+        self::assertEquals(Agent::class, get_class($agent));
+    }
+
+    public function testBuildAgentWithCollectors()
+    {
+        $this->collector
+            ->shouldReceive('useAgent')
+            ->shouldReceive('getName')
+            ->andReturn('span-collector');
```

Comment:
I'll sound like a broken record, but I think there should always be a `once` or `expects` to ensure the method is actually called. If it's not called then it's either not needed in the test or something doesn't work according to the expectation:

```suggestion
        $this->collector
            ->shouldReceive('useAgent')
                ->once()
            ->shouldReceive('getName')
                ->twice()
                ->andReturn('span-collector');
```

Reply:
I'm getting old and keep forgetting things like that.